utils/fetchMultiplePageResults.js

```diff
@@ -0,0 +1,22 @@
+const githubService = require("../services/githubService");
+
+const fetchMultiplePageResults = async (callbackFn, params) => {
+  let page = 1;
+  const { data } = await callbackFn({ page, ...params });
+
+  const allPRs = githubService.extractPRdetails(data);
+  page++;
+
+  while (allPRs.length === 100) {
+    const { data } = await callbackFn({ page, ...params });
+    const nextPRs = githubService.extractPRdetails(data);
+    allPRs.push(...nextPRs);
+    page++;
+  }
```

Comment:
cant it be called inside the loop itself? you can use a do-while loop also is it a good way to fetch in a loop?

Reply:
thanks! Updated the implementation using do while

Not really sure how else it can be done, we want to make requests sequentially instead of  making parallel requests.
Since the parent block of code is already wrapped in try catch, don't see a reason to wrap this with try catch unless we need to specifically return a particular error